src/main/java/com/bbloggsbott/readytoserve/pages/service/DirectoryWatchingService.java

```diff
@@ -0,0 +1,119 @@
+                logger.info("Error while monitoring {}", directory);
+                e.printStackTrace();
+            }
+            while (watchKey != null){
+                for (WatchEvent event: watchKey.pollEvents()){
+                    logger.info("Event: {}, File: {}", event.kind(), event.context());
+                    if (event.kind() == StandardWatchEventKinds.ENTRY_CREATE){
+                        String createdFile =  Paths.get(directory, event.context().toString()).toString();
+                        logger.info("Create Event detected for {}", createdFile);
+                        List<File> directoriesToWatch =  pagePathService.createPathsOnFileCreate(createdFile);
+                        watchDirectories(directoriesToWatch);
+                        logger.info("Create Event Processed");
+                    } else if (event.kind() == StandardWatchEventKinds.ENTRY_DELETE){
+                        String deletedFile = Paths.get(directory, event.context().toString()).toString();
+                        logger.info("Delete event detected for {}", deletedFile);
+                        pagePathService.deletePathsOnFileDelete(deletedFile);
+                        deleteDirectoryWatchingThread(deletedFile);
+                        logger.info("Delete event processed");
+                    } else if (event.kind() == StandardWatchEventKinds.ENTRY_MODIFY){
+                        String modifiedFile = Paths.get(directory, event.context().toString()).toString();
+                        logger.info("Modify event detected for {}", modifiedFile);
+                        pagePathService.modifyPathsOnFileModify(modifiedFile);
+                        logger.info("Modify Event processed");
+                    }
+                }
+                watchKey.reset();
+                try {
+                    watchKey = watchService.take();
+                } catch (InterruptedException e) {
+                    if (!new File(directoryPath.toString()).exists())
+                        break;
+                    logger.info("Error while monitoring {}", directory);
+                    e.printStackTrace();
+                }
+            }
+        };
+        Thread watcherThread = new Thread(watcher, directory);
+        watcherThreads.add(watcherThread);
+        watcherThread.start();
+    }
+
+    private void deleteDirectoryWatchingThread(String dirname){
+        if (!FilenameUtils.getExtension(dirname).isEmpty()){
+            return;
+        }
+        logger.info("Stopping watcher thread for {}", dirname);
+        watcherThreads = watcherThreads.stream().filter(thread -> {
+            if (thread.getName().equalsIgnoreCase(dirname)){
+                thread.interrupt();
+                return false;
+            }
+            return true;
+        }).collect(Collectors.toList());
+        logger.info("Stopped watcher thread for {}", dirname);
+    }
+
+    public void watchDirectories(List<File> directories) {
```

Comment:
`@Async` Here could have saved you from writing a lot of code.

Reply:
I also want to stop the thread when the directory is deleted. Will `@Async` let me do that?